contracts/src/v0.8/operatorforwarder/dev/OperatorFactory.sol

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {Operator} from "./Operator.sol";
+import {AuthorizedForwarder} from "./AuthorizedForwarder.sol";
+
+/**
+ * @title Operator Factory
+ * @notice Creates Operator contracts for node operators
+ */
+ // solhint-disable custom-errors
+contract OperatorFactory {
+  // solhint-disable-next-line chainlink-solidity/prefix-immutable-variables-with-i
+  address public immutable getChainlinkToken;
+  mapping(address => bool) private s_created;
+
+  event OperatorCreated(address indexed operator, address indexed owner, address indexed sender);
+  event AuthorizedForwarderCreated(address indexed forwarder, address indexed owner, address indexed sender);
+
+  /**
+   * @param linkAddress address
+   */
+  constructor(address linkAddress) {
+    getChainlinkToken = linkAddress;
+  }
+
+  /**
+   * @notice The type and version of this contract
+   * @return Type and version string
+   */
+  function typeAndVersion() external pure virtual returns (string memory) {
+    return "OperatorFactory 1.0.0";
+  }
+
+  /**
+   * @notice creates a new Operator contract with the msg.sender as owner
+   */
+  function deployNewOperator() external returns (address) {
+    Operator operator = new Operator(getChainlinkToken, msg.sender);
+
+    s_created[address(operator)] = true;
+    emit OperatorCreated(address(operator), msg.sender, msg.sender);
+
+    return address(operator);
+  }
+
+  /**
+   * @notice creates a new Operator contract with the msg.sender as owner and a
+   * new Operator Forwarder with the Operator as the owner
+   */
+  function deployNewOperatorAndForwarder() external returns (address, address) {
+    Operator operator = new Operator(getChainlinkToken, msg.sender);
+    s_created[address(operator)] = true;
+    emit OperatorCreated(address(operator), msg.sender, msg.sender);
+
+    bytes memory tmp = new bytes(0);
```

Comment:
can inline to save gas (more cases below)

Reply:
done both accts